### ISSUE TITLE
Stacking results fails with odd number of frames

### ISSUE THREAD

**Issue by mimischi** [MDAnalysis/pmda#58]
I've got a trajectory (7367 frames) that was split across four CPUs, resulting in partial trajectories with 3x 1842 and 1x 1841 frames. Running `np.hstack(self._results)` does not work, it should either be `np.vstack` or `np.concatenate`.

https://github.com/MDAnalysis/pmda/blob/723d02163a20991b712910296b9f39595ddc9826/pmda/custom.py#L102-L103

### Expected behaviour

Results are concatenated over the frame axis.

### Actual behaviour

`ValueError` is raised.

### Code to reproduce the behaviour

```python
import numpy as np
data = np.zeros([7367, 10])
print(data.shape)
# (7367, 10)

data_frames = np.array_split(data, 4)
print([frames.shape[0] for frames in data_frames])
# [1842, 1842, 1842, 1841]

np.hstack(data_frames)
# [...]
# ValueError: all the input array dimensions except for the concatenation axis must match exactly

np.vstack(data_frames)
# array([[0., 0., 0., ..., 0., 0., 0.],
#       [0., 0., 0., ..., 0., 0., 0.],
#       [0., 0., 0., ..., 0., 0., 0.],
#       ...,
#       [0., 0., 0., ..., 0., 0., 0.],
#       [0., 0., 0., ..., 0., 0., 0.],
#       [0., 0., 0., ..., 0., 0., 0.]])
```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.18.0

(run `python -c "import pmda; print(pmda.__version__)"`)
0.1.1

(run `python -c "import dask; print(dask.__version__)"`)
0.19.1

**Comment by kain88-de** [comment]
This is currently not noticed in testing because we only run the tests on a single core. See #57

**Comment by orbeckst** [comment]
I guess I found the same a while back in `rms.py` https://github.com/MDAnalysis/pmda/blob/a5aa57607bc413496322d97bbc22e8eb64884cba/pmda/rms.py#L76

**Comment by mimischi** [comment]
I could fix this bug, if you want to. Would a PR also need to resolve #57 at the same time?

**Comment by kain88-de** [comment]
It doesn't need to. I won't complain if it does.

**Comment by orbeckst** [comment]
PR, pretty-please!

(tests are probably the major issue; we do have a few tests that use two workers)
